sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java

```diff
@@ -315,32 +318,49 @@ public String apply(@Nullable ExtraContextInfo input) {
     return contextInfos;
   }
 
+  /** Interface for invoking the {@code DoFn} processing methods. */
+  public interface DoFnInvoker<InputT, OutputT>  {
+    /** Invoke {@link DoFn#startBundle} on the bound {@code DoFn}. */
+    void invokeStartBundle(
+        DoFnWithContext<InputT, OutputT>.Context c,
+        ExtraContextFactory<InputT, OutputT> extra);
+    /** Invoke {@link DoFn#finishBundle} on the bound {@code DoFn}. */
+    void invokeFinishBundle(
+        DoFnWithContext<InputT, OutputT>.Context c,
+        ExtraContextFactory<InputT, OutputT> extra);
+
+    /** Invoke {@link DoFn#processElement} on the bound {@code DoFn}. */
+    public void invokeProcessElement(
+        DoFnWithContext<InputT, OutputT>.ProcessContext c,
+        ExtraContextFactory<InputT, OutputT> extra);
+  }
+
   /**
    * Implementation of {@link DoFnReflector} for the arbitrary {@link DoFnWithContext}.
```

Comment:
Is this more than one implementation? Is `DoFnReflector` now just a `DoFnInvokerFactory`?